sklearn/preprocessing/data.py

```diff
@@ -2512,25 +2515,22 @@ def _fit(self, X, y=None, force_compute_transform=False):
                 self.lambdas_.append(lmbda)
         self.lambdas_ = np.array(self.lambdas_)
 
-        transformed = []
-        if self.standardize or force_compute_transform:
+        if self.standardize or force_transform:
             transform_function = {'box-cox': boxcox,
                                   'yeo-johnson': self._yeo_johnson_transform
                                   }[self.method]
-            for col, lmbda in zip(X.T, self.lambdas_):
+            for i, lmbda in enumerate(self.lambdas_):
                 with np.errstate(invalid='ignore'):  # hide NaN warnings
-                    col_trans = transform_function(col, lmbda)
-                    transformed.append(col_trans)
-            transformed = np.array(transformed).T
+                    X[:, i] = transform_function(X[:, i], lmbda)
 
         if self.standardize:
-            self._scaler = StandardScaler()
-            if force_compute_transform:
-                transformed = self._scaler.fit_transform(transformed)
+            self._scaler = StandardScaler(copy=self.copy)
```

Comment:
actually you should be able to use `copy=False` here, since a copy has already been done just before.